api/swagger.yaml

```diff
@@ -8339,6 +8339,8 @@ paths:
                   UsageData:
                     Size: 10920104
                     RefCount: 2
+              BuildCache: null
```

Comment:
Not for this PR (probably), but we should change this to return an empty slice instead of `nil`. On a fresh daemon, this shows:

```bash
curl -s --unix-socket /var/run/docker.sock http://localhost/system/df | jq .

```json
{
  "LayersSize": 0,
  "Images": [],
  "Containers": [],
  "Volumes": [],
  "BuildCache": null,
  "BuilderSize": 0
}
```

To be consistent, I think it makes sense to return an empty slice instead of `nil`

Reply:
Perhaps we should add an actual example here (but we can keep that separately, I'll check with you on Slack)

```suggestion
              BuildCache:
                -
                  ID: "hw53o5aio51xtltp5xjp8v7fx",
                  Parent: "",
                  Type: "regular",
                  Description: "pulled from docker.io/library/debian@sha256:234cb88d3020898631af0ccbbcca9a66ae7306ecd30c9720690858c1b007d2a0",
                  InUse: false,
                  Shared: true,
                  Size: 0,
                  CreatedAt: "2021-06-28T13:31:01.474619385Z",
                  LastUsedAt: "2021-07-07T22:02:32.738075951Z",
                  UsageCount: 26
                -
                  ID: "ndlpt0hhvkqcdfkputsk4cq9c",
                  Parent: "hw53o5aio51xtltp5xjp8v7fx",
                  Type: "regular",
                  Description: "mount / from exec /bin/sh -c echo 'Binary::apt::APT::Keep-Downloaded-Packages \"true\";' > /etc/apt/apt.conf.d/keep-cache",
                  InUse: false,
                  Shared: true,
                  Size: 51,
                  CreatedAt: "2021-06-28T13:31:03.002625487Z",
                  LastUsedAt: "2021-07-07T22:02:32.773909517Z",
                  UsageCount: 26
```

Reply:
See https://github.com/moby/moby/pull/42607. Included your example in a separate commit as well.